compiler/src/dotty/tools/dotc/ast/untpd.scala

```diff
@@ -525,8 +525,8 @@ object untpd extends Trees.Instance[Untyped] with UntypedTreeInfo {
   def makeRetaining(parent: Tree, refs: List[Tree], annotName: TypeName)(using Context): Annotated =
     Annotated(parent, New(scalaAnnotationDot(annotName), List(refs)))
 
-  def makeCapsOf(id: Ident)(using Context): Tree =
-    TypeApply(Select(scalaDot(nme.caps), nme.capsOf), id :: Nil)
+  def makeCapsOf(tp: Tree)(using Context): Tree =
```

Comment:
I extended it so it can also accept type member of a path:

```scala
trait Source[T]:
  type Cap >: caps.CapSet^{this} <: caps.CapSet^

def race[T](a: Source[T]^, b: Source[T]^) = new Source[T]:
  type Cap = caps.CapSet^{this, a.Cap^, b.Cap^}
```

Reply:
Ah OK. Then I guess we can narrow to RefTrees.